signer/src/eth.rs

```diff
@@ -0,0 +1,269 @@
+// Copyright 2019-2023 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+//! An ethereum signer implementation.
+use derive_more::{Display, From};
+use hex::FromHex;
+use keccak_hash::keccak;
+use secp256k1::{Keypair, Message, Secp256k1, SecretKey};
+
+/// An ethereum signer implementation.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct EthereumSigner(Keypair);
```

Comment:
I would prefer to call this `Signer` or `KeyPair` because it's already in module named `eth` and the `Ethereum` prefix can be omitted.

Reply:
can we wrap the `ecdsa::KeyPair` here instead?